scrapy/core/engine.py

```diff
@@ -136,7 +136,9 @@ def unpause(self) -> None:
         self.paused = False
 
     def _next_request(self) -> None:
-        assert self.slot is not None  # typing
+        if self.slot is None:
+            return
```

Comment:
This is necessary because `close_spider` uses `setattr` to set `self.slot` to `None`. However, I wonder if we can m,ake it so that `_next_request` in not called in the first place, in those scenarios.